### PR TITLE
config-sync: adding initial user documentation for this feature

### PR DESCRIPTION
**Pull request by Giggum**
## Change Summary

User documentation created for the config-sync service.
Adds a new page called "Config Sync" under Configuration Guide > Service

## Related Task(s)

* https://vyos.dev/T775
* https://vyos.dev/T6121

## Backport

Sagitta

## Checklist:

- [X] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-documentation/blob/current/CONTRIBUTING.md) document

### CODE REVIEW

File: docs/configuration/service/config-sync.rst [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+be synchronized and the mode to use - whether to replace or add. 
+
+To prevent issues with divergent configurations between the pair of routers, 
+synchronization is strictly unidirectional from primary to replica. Both 
+routers should be online and run the same version of VyOS.
+
+Configuration
+-------------
+
+.. cfgcmd:: set service config-sync section secondary
```
Comment (sever-sever) [inline docs/configuration/service/config-sync.rst]: “section secondary” seems incorrect config

File: docs/configuration/service/config-sync.rst [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+Example
+-------
+* Synchronize the time-zone and OSPF configuration from Router A to Router B
+* The address of Router B is 10.0.20.112 and the port used is 8443
+
+Configure the HTTP API service on Router B
+
+.. code-block:: none
+
+    set service https api graphql authentication expiration '31536000'
```
Comment (sever-sever) [inline docs/configuration/service/config-sync.rst]: Graphql is not required

File: docs/configuration/service/config-sync.rst (hunk 1 above)
Comment (Giggum) [reply docs/configuration/service/config-sync.rst]: Thank you for catching this, will correct.

File: docs/configuration/service/config-sync.rst (hunk 2 above)
Comment (Giggum) [reply docs/configuration/service/config-sync.rst]: Thank you, will omit and retest without to ensure.

Comment (Giggum) [reply docs/configuration/service/config-sync.rst]: @sever-sever confirming it works with just a basic HTTPS-API config see below. I've modified the documentation as you suggested.

```
vyos@vyos-B# delete service https api graphql 
vyos@vyos-B# commit
vyos@vyos-B# run show configuration commands | match "https"
set service https api keys id KID key 'foo'
set service https listen-address '10.0.20.113'
set service https port '8443'
```

**Router A**
```
% ssh vyos@10.0.20.111
vyos@10.0.20.111's password: 
Welcome to VyOS!

   ┌── ┐
   . VyOS 1.5-rolling-202405060019
   └ ──┘  current

 * Documentation:  https://docs.vyos.io/en/latest
 * Project news:   https://blog.vyos.io
 * Bug reports:    https://vyos.dev

You can change this banner using "set system login banner post-login" command.

VyOS is a free software distribution that includes multiple components,
you can check individual component licenses under /usr/share/doc/*/copyright
Last login: Tue May  7 16:42:28 2024 from 10.0.20.71
vyos@vyos-A:~$ conf
vyos@vyos-A# set system time-zone 'Europe/Amsterdam'
vyos@vyos-A# commit
INFO:vyos_config_sync:Config synchronization: Mode=load, Secondary=10.0.20.113
vyos@vyos-A# set protocols ospf area 0 network '10.111.222.0/30'
vyos@vyos-A# commit
INFO:vyos_config_sync:Config synchronization: Mode=load, Secondary=10.0.20.113
```

**Router B**
```
% ssh vyos@10.0.20.113
vyos@10.0.20.113's password: 
Welcome to VyOS!

   ┌── ┐
   . VyOS 1.5-rolling-202405060019
   └ ──┘  current

 * Documentation:  https://docs.vyos.io/en/latest
 * Project news:   https://blog.vyos.io
 * Bug reports:    https://vyos.dev

You can change this banner using "set system login banner post-login" command.

VyOS is a free software distribution that includes multiple components,
you can check individual component licenses under /usr/share/doc/*/copyright
Last login: Tue May  7 13:35:23 2024 from 10.0.20.71
vyos@vyos-B:~$ show configuration commands | match "time-zone"
set system time-zone 'Europe/Amsterdam'
vyos@vyos-B:~$ show configuration commands | match "ospf"
set protocols ospf area 0 network '192.0.2.0/30'
set protocols ospf area 0 network '10.111.222.0/30'
```

**Review by sever-sever — APPROVED** [review]

**Comment by rebortg** [comment]
@mergifyio backport sagitta

**Pull request merged**